src/app/filelogger.cpp

```diff
@@ -68,26 +70,21 @@ FileLogger::~FileLogger()
     closeLogFile();
 }
 
-void FileLogger::changePath(const QString &newPath)
+void FileLogger::changePath(const Path &newPath)
 {
-    const QDir dir(newPath);
-    dir.mkpath(newPath);
-    const QString tmpPath = dir.absoluteFilePath("qbittorrent.log");
+    Utils::Fs::mkpath(newPath);
 
-    if (tmpPath != m_path)
-    {
-        m_path = tmpPath;
+    m_path = newPath / Path("qbittorrent.log");
 
-        closeLogFile();
-        m_logFile.setFileName(m_path);
-        openLogFile();
-    }
+    closeLogFile();
```

Comment:
> And maybe keep the `if (tmpPath != m_path)` comparison?

Then we have to compare them in "case sensitive" way, otherwise user won't be able to change only case of some path element.

Reply:
> I feel unsafe when reading this part, can you refactor it more to:

Done.